indy-repository-manager/src/main/java/org/jboss/pnc/indyrepositorymanager/IndyRepositorySession.java

```diff
@@ -710,6 +719,95 @@ private String computeIdentifier(final TrackedContentEntryDTO transfer) {
         return identifier;
     }
 
+    /**
+     * Computes purl string for an artifact.
+     *
+     * @param transfer the download or upload that we want to generate identifier for
+     * @return generated purl
+     */
+    private String computePurl(final TrackedContentEntryDTO transfer) {
+        String purl = null;
+
+        try {
+            switch (transfer.getStoreKey().getPackageType()) {
+                case MAVEN_PKG_KEY:
+
+                    ArtifactPathInfo pathInfo = ArtifactPathInfo.parse(transfer.getPath());
+                    if (pathInfo != null) {
+                        // See https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#maven
+                        PackageURLBuilder purlBuilder = PackageURLBuilder.aPackageURL()
+                                .withType(PackageURL.StandardTypes.MAVEN)
+                                .withNamespace(pathInfo.getProjectId().getGroupId())
+                                .withName(pathInfo.getProjectId().getArtifactId())
+                                .withVersion(pathInfo.getVersion())
+                                .withQualifier(
+                                        "type",
+                                        StringUtils.isEmpty(pathInfo.getType()) ? "jar" : pathInfo.getType());
```

Comment:
This is probably an unnecessary default, because when you parse the ArtifactPathInfo from path, it always has a type. If it is not there, the parser is not able to complete and returns null.

Reply:
Ah ok, I took the hint from SimpleTypeAndClassifier constructor (used by SimpleArtifactRef which is used in the creation of the identifier)